x/concentrated-liquidity/fees.go

```diff
@@ -285,7 +287,7 @@ func computeFeeChargePerSwapStep(currentSqrtPrice, nextTickSqrtPrice, sqrtPriceL
 	// In both cases, charge fee on the full amount that the tick
 	// originally had.
 	if didReachNextSqrtPrice || isPriceImpactProtection {
-		feeChargeTotal = amountIn.Mul(swapFee)
+		feeChargeTotal = math.MulRoundUp(amountIn, swapFee)
```

Comment:
just curious, what led us to the decision to round this up?

Reply:
This is pretty minor but just a safety precaution.

Default `Mul` does banker's rounding. However, we want to always round up the fee so that we don't undercharge it

Reply:
makes sense, can we have in line doc regarding this on this line on why we mul round up here? I think it would be useful context for future code mangement